packages/gatsby/src/bootstrap/load-config-and-plugins.ts

```diff
@@ -0,0 +1,94 @@
+import reporter from "gatsby-cli/lib/reporter"
+import telemetry from "gatsby-telemetry"
+
+import { IFlattenedPlugin } from "./load-plugins/types"
+
+import { preferDefault } from "../bootstrap/prefer-default"
+import { getConfigFile } from "../bootstrap/get-config-file"
+import { loadPlugins } from "../bootstrap/load-plugins"
+import { internalActions } from "../redux/actions"
+import loadThemes from "../bootstrap/load-themes"
+import { store } from "../redux"
+import handleFlags from "../utils/handle-flags"
+import availableFlags from "../utils/flags"
+
+export async function loadConfigAndPlugins({
+  siteDirectory,
+  processFlags = false,
+}: {
+  siteDirectory: string
+  processFlags: boolean
+}): Promise<{
+  config: any
```

Comment:
I guess we can't type `config` at the moment?

Reply:
Maybe we could, but requires more effort that is "tangential" to this PR (IMO) - it's `any` currently in master branch too ( `configModule` type in https://github.com/gatsbyjs/gatsby/blob/db946a564630f2d9fdeddea78d6d57e2f6d72bc4/packages/gatsby/src/bootstrap/get-config-file.ts#L17-L24 )

We do have config type for what we store in redux - https://github.com/gatsbyjs/gatsby/blob/db946a564630f2d9fdeddea78d6d57e2f6d72bc4/packages/gatsby/src/redux/types.ts#L60-L82
but this doesn't contain for example `flags` (I did look to have it type briefly, but saw there is lot of unrelated things to handle, so decided to not pursue it for this PR)